### ISSUE TITLE
Support for staggered start

### ISSUE THREAD

**Issue by anthonyryan1** [lilydjwg/pssh#142]
This is a pretty unique use case, but it would be nice to have a staggered start without forcing a low parallel value.

My workstation uses a TPM SSH key to connect to remote hosts: https://wiki.gentoo.org/wiki/Trusted_Platform_Module/SSH
My TPM only supports 12 requests in parallel.

This means my desktop has a limit of 12 new SSH connections at any one time, but no limit on how many established SSH connections are running.

I can mitigate this by using `-p 12`, but if I'm running long SSH commands it would be nice to stagger the startup without losing the parallelism across a few hundred machines.

**Comment by lilydjwg** [comment]
But how do you tell when the "start" finishes?

OpenSSH has a `ControlMaster` feature that can persist connections across different command calls. Perhaps you can write a script to establish the connections and then use `pssh` to do the actual task? A configuration like below in your `~/.ssh/config` should work:

```
ControlPath /run/user/1000/ssh/master-%r@%h:%p
ControlMaster auto
ControlPersist yes
```

**Comment by anthonyryan1** [comment]
I've given it some thought, and I'm failing to come up with a better solution than your suggested workaround.

1. We can't look for output to stderr or stdout to know we're connected, because not all commands will provide any output there.
2. SSH in verbose mode would be a lot of extra noise that would impede uses of the `--print` argument.
3. Using something like Paramkio would let us know when the connection has been made, but it's definitely going to be missing features that OpenSSH will have and that will break functionality someone depends on
    - Pretty sure this includes my TPM backed private key

For lack of a better solution, I'm going to close this issue and use your workaround until there's a better idea.
